src/transforms/stdnames.jl

```diff
@@ -0,0 +1,61 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    StdNames(:spec)
+
+StdNames transform standardizes the column names.
+"""
+struct StdNames <: Stateless
+  spec::Symbol
+end
+
+StdNames() = StdNames(:upper)
+
+isrevertible(::Type{StdNames}) = true
+
+function apply(transform::StdNames, table)  
+  cols = Tables.columns(table)
+  oldnames = Tables.columnnames(cols)
+  
+  spec = transform.spec
+
+  spec == :camel && (newstringnames = _camel.(string.(oldnames)))
+  spec == :snake && (newstringnames = _snake.(string.(oldnames)))
+  spec == :upper && (newstringnames = _upper.(string.(oldnames)))
+
+  newuniquenames = _unique(newstringnames)
+  newnames = Symbol.(newuniquenames)
+  names = Dict(zip(oldnames, newnames))
+
+  rtrans = Rename(names)
+  newtable, rcache = apply(rtrans, table)
+
+  newtable, (rtrans, rcache)
+end
+
+function revert(::StdNames, newtable, cache)
+  rtrans, rcache = cache
+  revert(rtrans, newtable, rcache)
+end
+
+const delim = ['_', ' ']
+
+function _unique(names)
+  uniquenames = [names...]
+  for i in range(2,length(uniquenames))
+      prev = uniquenames[1:i-1]
+      while uniquenames[i] in prev
+        uniquenames[i] = string(uniquenames[i], "_")
+      end
+  end
+
+  Tuple(uniquenames)
+end
```

Comment:
Again, you are mixing 2-space indentation with 4-space indentation. Please use an IDE and set this to avoid bumping into this issue again? Also, the implementation is not great. Simply create an empty vector of Symbol and keep pushing to it if the entry is not already in the vector. Don't use indexing.

Reply:
Is this what you had in mind? I don't think our vector needs to be `Vector{Symbol}`, as we are working with strings.
```julia
function unique(names)
  newnames = Vector{String}()
  for name in names
    updatedname = name
    while updatedname ∈ newnames
      updatedname = string(updatedname, "_")
    end
    push!(newnames, updatedname)
  end

  newnames
end
```